src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -732,11 +732,25 @@ private void parseConnections(final List<Element> connections) throws GameParseE
     }
   }
 
-  private void parseResources(final Element root) {
-    getChildren("resource", root).stream()
-        .map(e -> e.getAttribute("name"))
-        .map(name -> new Resource(name, data))
-        .forEach(data.getResourceList()::addResource);
+  private void parseResources(final Element root) throws GameParseException {
+    for (final Element element : getChildren("resource", root)) {
+      final String name = element.getAttribute("name");
+      final String isDisplayedFor = element.getAttribute("isDisplayedFor");
+      final List<PlayerID> players = data.getPlayerList().getPlayers();
+      if (isDisplayedFor.equalsIgnoreCase("NONE")) {
```

Comment:
nit, magic constant `"NONE"`. Looking at the rest of this update, it looks to be the default missing value supplied by XML parser. If someone were looking at this code without looking at this PR, that may be very unclear where `NONE` is coming from and whether it was user supplied in the XML. A well named constant could help avoid that confusion.

Reply:
Not sure a constant at the top of a huge class is any better. Really the parser class needs chunked up and then a constant would help more.

Reply:
Added constant.